install/index.php

```diff
@@ -1864,34 +1862,14 @@ function revertSetting(defval, inpid)
 		{
 			$hostname .= ':'.$_SERVER['SERVER_PORT'];
 		}
-		$websiteurl = $hostname.'/';
-
+		
 		$currentlocation = get_current_location();
-		if($currentlocation)
-		{
-			// TODO: Change this to find the last position of /install/
-			$pos = my_strpos($currentlocation, '/install/');
-			if($pos === 0)
-			{
-				$cookiepath = "/";
-			}
-			else
-			{
-				$cookiepath = my_substr($currentlocation, 0, $pos).'/';
-			}
-		}
-
-		$currentscript = $hostname.get_current_location();
-
-		if($currentscript)
-		{
-			$bburl = my_substr($currentscript, 0, my_strpos($currentscript, '/install/'));
-		}
-
-		if($_SERVER['SERVER_ADMIN'])
-		{
-			$contactemail = $_SERVER['SERVER_ADMIN'];
-		}
+		$noinstall = substr($currentlocation, 0, strrpos($currentlocation, '/install/'));
```

Comment:
Oh, forgot to answer here. Anyways: http://stackoverflow.com/questions/708017/can-a-php-file-name-or-a-dir-in-its-full-path-have-utf-8-characters?answertab=votes#tab-top - don't know how up-to-date it is, but multi-byte characters are not recommended for file/fiolder names and may break more things in MyBB.

And even if someone provides the folder you said, once again, you can test it:
```php
$footer .= substr('/téstèfõldër/install/index.php', 0, strrpos('/téstèfõldër/install/index.php', '/install/')).'/';
$footer .= '<br />';
$footer .= my_substr('/téstèfõldër/install/index.php', 0, strrpos('/téstèfõldër/install/index.php', '/install/')).'/';
```
Results in:
```php
/téstèfõldër/
/téstèfõldër/ins/
```
So my version is correct and using `my_substr() breaks it. Why? Because the last occurence's position is calculated without considering multi-byte characters and same goes for `substr()`, thus the output is good. `my_substr()` on the other hands counts the characters differently and won't work properly unless we define `my_strrpos()` too and use it instead of `strrpos()`.